src/ipc/ipc-helper.c

```diff
@@ -173,13 +174,9 @@ int comp_buffer_connect(struct comp_dev *comp, uint32_t comp_core,
 			struct comp_buffer *buffer, uint32_t dir)
 {
 	/* check if it's a connection between cores */
-	if (buffer->core != comp_core) {
-		/* set the buffer as a coherent object */
-		coherent_shared_thread(buffer, c);
-
+	if (buffer->core != comp_core)
```

Comment:
do we need add:
buffer->is_shared = true;
or check it here?

Reply:
checking makes sense
adding buffer->is_shared = true;  is the very bad idea as it won't change memory alias the buffer is accessed by